src/components/stats.jsx

```diff
@@ -0,0 +1,231 @@
+    const height = +svg.attr('height') - margin.top - margin.bottom;
+    const g = svg.append('g').attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+    const x = d3Scale.scaleBand()
+        .rangeRound([0, width])
+        .padding(0.1)
+        .align(0.1);
+
+    const y = d3Scale.scaleLinear()
+        .rangeRound([height, 0]);
+
+    const z = d3Scale.scaleOrdinal()
+        .range(['#d0743c', '#98abc5']);
+
+    const stack = d3Shape.stack();
+
+    x.domain(data.map(d => d.version));
+    y.domain([0, d3Array.max(data, d => d.total)]).nice();
+    z.domain(columns);
+
+    g.selectAll('.serie')
+      .data(stack.keys(columns)(data))
+      .enter().append('g')
+        .attr('class', 'serie')
+        .attr('fill', d => z(d.key))
+      .selectAll('rect')
+      .data(d => d)
+      .enter().append('rect')
+        .attr('x', d => x(d.data.version))
+        .attr('y', d => y(d[1]))
+        .attr('height', d => y(d[0]) - y(d[1]))
+        .attr('width', x.bandwidth());
+
+    g.append('g')
+        .attr('class', 'axis axis--x')
+        .attr('transform', `translate(0, ${height})`)
+        .call(d3Axis.axisBottom(x));
+
+    let constraintText = '';
+
+    switch (probeConstraint) {
+      case 'new_in': constraintText = 'new'; break;
+      case 'is_in': constraintText = 'recorded'; break;
+      case 'is_expired': constraintText = 'expired'; break;
+      default: console.error('Unknown version constraint.');
+    }
+
+    g.append('g')
+        .attr('class', 'axis axis--y')
+        .call(d3Axis.axisLeft(y).ticks(10, 's'))
+      .append('text')
+        .attr('x', 2)
+        .attr('y', y(y.ticks(10).pop()))
+        .attr('dy', '0.35em')
+        .attr('text-anchor', 'start')
+        .attr('fill', '#000')
+        .text(`Count of ${constraintText} probes`);
+
+    const legendLabels = [];
+
+    if (hasOptinData) {
+      legendLabels.push('optin');
+    }
+    if (hasOptoutData) {
+      legendLabels.push('optout');
+    }
+
+    const legend = g.selectAll('.legend')
+      .data(legendLabels.reverse())
+      .enter().append('g')
+        .attr('class', 'legend')
+        .attr('transform', (d, i) => `translate(0, ${i * 20})`)
+        .style('font', '10px sans-serif');
+
+    legend.append('rect')
+        .attr('x', width - 18)
+        .attr('width', 18)
+        .attr('height', 18)
+        .attr('fill', z);
+
+    legend.append('text')
+        .attr('x', width - 24)
+        .attr('y', 9)
+        .attr('dy', '.35em')
+        .attr('text-anchor', 'end')
+        .text(d => d);
+  }
+
+  render() {
+    const classNames = ['tab-pane', 'active'];
+    if (this.props.activeView !== 'stats') classNames.push('hidden');
+
+    return (
+      <div className={classNames.join(' ')} id="stats-view" aria-expanded="true">
+        <svg className="img-fluid" id="stats-content" width="960" height="500" />
+      </div>
+    );
+  }
+}
+
+export default Stats;
```

Comment:
Do you think it would be over-optimizing to split this file up? It's pretty big, but I can be convinced otherwise.